glossary.md

```diff
@@ -0,0 +1,241 @@
+# Glossary
+
+--------------------------------------------------------------------------------
+
+# General Terms
+
+## Layer 1 (L1)
+[L1]: /glossary.md#layer-1-L1
+
+Refers the Ethereum blockchain, used used in contrast to [layer 2][L2], which
+refers to Optimistic Ethereum.
+
+## Layer 2 (L2)
+[L2]: /glossary.md#layer-2-L2
+
+Refers to the Optimistic Ethereum blockchain (specified in this repository),
+used in contrast to [layer 1][L1], which refers to the Ethereum blockchain.
+
+## Block
+[block]: /glossary.md#layer-1-L1
+
+Can refer to an [L1] block, or to an [L2] block, which are structured similarly.
+
+A block is a sequential list of transactions, along with a couple of properties
+stored in the *header* of the block. A description of these properties can be
+found in code comments [here][nano-header], or in the [Ethereum yellow paper
```

Comment:
It's a bit overkill since this is the glossary. The goal wasn't to be specificative necessarily but informative.
But you're right that there is the potential to fall behind, and it's confusing because, well this is a specification.

I'm not too sure what to do in this context. Listing everything like you say is not the goal here. I just wanted to give the reader a quick idea of what's in a header, and that seemed the easiest way to do it. Got any thoughts?

That being said, the problem of precise specification will be present in the execution engine spec. I'm thinking we shoud link to the yellow paper + the front page of the execution-specs repo that has the whole EIP list per hard fork, does that seem alright?

Reply:
discussion here: https://github.com/ethereum-optimism/optimistic-specs/issues/46